src/kernel/src/regmgr/mod.rs

```diff
@@ -299,6 +310,28 @@ impl RegMgr {
 
         None
     }
+
+    fn sys_workaround8849(self: &Arc<Self>, i: &SysIn) -> Result<SysOut, SysErr> {
+        let key = {
+            let arg: usize = i.args[0].into();
+            let key: u32 = arg.try_into().unwrap();
+
+            RegKey::new(key)
+        };
+
+        match key {
+            RegKey::NP_DEBUG
+            | RegKey::BROWSER_DEBUG_NOTIFICATION
+            | RegKey::DEVENV_TOOL_TRC_NOTIFY
+            | RegKey::DEVENT_TOOL_USE_DEFAULT_LIB
+            | RegKey::DEVENV_TOOL_SYS_PRX_PRELOAD => {
+                let val = self.get_int(key).unwrap();
+
+                Ok(val.into())
```

Comment:
`td_retval` will be ignored if non-zero is returned from the syscall handler. See https://github.com/freebsd/freebsd-src/blob/release/9.1.0/sys/amd64/amd64/vm_machdep.c#L380 for how it is handled.

Reply:
Thanks, I'll look into it when I have some time.